irohad/multi_sig_transactions/transport/impl/mst_transport_grpc.cpp

```diff
@@ -3,70 +3,87 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include "common/default_constructible_unary_fn.hpp"  // non-copyable value workaround
+
 #include "multi_sig_transactions/transport/mst_transport_grpc.hpp"
 
-#include "backend/protobuf/transaction.hpp"
-#include "builders/protobuf/transport_builder.hpp"
-#include "interfaces/iroha_internal/transaction_batch.hpp"
-#include "interfaces/iroha_internal/transaction_sequence_factory.hpp"
+#include <boost/range/adaptor/filtered.hpp>
+#include "interfaces/iroha_internal/transaction_batch_factory.hpp"
 #include "validators/default_validator.hpp"
-#include "validators/transactions_collection/batch_order_validator.hpp"
 
 using namespace iroha::network;
 
 MstTransportGrpc::MstTransportGrpc(
     std::shared_ptr<network::AsyncGrpcClient<google::protobuf::Empty>>
         async_call,
-    std::shared_ptr<shared_model::interface::CommonObjectsFactory> factory)
-    : async_call_(std::move(async_call)), factory_(std::move(factory)) {}
+    std::shared_ptr<shared_model::interface::CommonObjectsFactory> factory,
+    std::shared_ptr<TransportFactoryType> transaction_factory,
+    std::shared_ptr<shared_model::interface::TransactionBatchParser>
+        batch_parser)
+    : async_call_(std::move(async_call)),
+      factory_(std::move(factory)),
+      transaction_factory_(std::move(transaction_factory)),
+      batch_parser_(std::move(batch_parser)) {}
+
+shared_model::interface::types::SharedTxsCollectionType
+MstTransportGrpc::deserializeTransactions(
+    const ::iroha::network::transport::MstState *request) {
+  return boost::copy_range<
+      shared_model::interface::types::SharedTxsCollectionType>(
+      request->transactions()
+      | boost::adaptors::transformed(
+            [&](const auto &tx) { return transaction_factory_->build(tx); })
+      | boost::adaptors::filtered([&](const auto &result) {
+          return result.match(
+              [](const iroha::expected::Value<
```

Comment:
It will, but is it necessary? When a reader will see `auto` here, their though will be "OK, this branch might be either value or error", and only later they will see the second branch and deduce the previous branch.